### PR TITLE
Honor freeze_automation in group.yml for ocp4

### PR DESCRIPTION
**Pull request by jupierce**
https://github.com/openshift/doozer/pull/121 must merge first.

This allows you to set builds.permitted in group.yml to either 'none', 'any', or 'manual'. If `none`, the pipeline will now exit with UNSTABLE and not make any attempt to modify distgit  or build. If `manual`, the pipeline will verify that a human executed the job (directly or indirectly via propagation).

### CODE REVIEW

**Comment by tbielawa** [comment]
This seems legit to me
/lgtm

File: jobs/build/ocp4/Jenkinsfile [hunk 1]
```diff
@@ -119,6 +119,13 @@ node {
         }
         stage("report success") { build.stageReportSuccess() }
     } catch (err) {
+
+        if ( buildPlan.is_permitted == false ) {
+            echo 'Exiting because this build is not permitted: ${err}'
```
Comment (sosiouxme) [inline jobs/build/ocp4/Jenkinsfile]: Could it go in the `currentBuild.description` too?

File: jobs/build/ocp4/build.groovy [hunk 2]
```diff
@@ -240,6 +243,33 @@ def planBuilds() {
         currentBuild.description += "error during plan builds step:\n${err.getMessage()}\n"
         throw err
     }
+
+    // check whether the group should be built right now
```
Comment (sosiouxme) [inline jobs/build/ocp4/build.groovy]: This is OK as far as it goes, but can we extract this to commonlib and use it for the `sweep` job as well? That's another thing we don't want running during release work.

File: jobs/build/ocp4/build.groovy [hunk 3]
```diff
@@ -240,6 +243,33 @@ def planBuilds() {
...
+
+    echo "Group's freeze_automation flag: ${buildPlan.freeze_automation}"
+    echo "Builder email: ${buildPlan.builderEmail}"
+
+    if (buildPlan.freeze_automation == "yes") {
+        echo "All automation are currently disabled in group.yml."
+        buildPlan.is_permitted = false
+    }
+
+    if (buildPlan.freeze_automation == "manual" && buildPlan.builderEmail == null) {
```
Comment (vfreex) [inline jobs/build/ocp4/build.groovy]: The `builderEmail` part is like a hack. It doesn't donate a `-1` but would be better to handle this situation better.

Comment (jupierce) [reply jobs/build/ocp4/build.groovy]: Unfortunately, this is the most elegant way I know of given the current plugins. Happy to be proven wrong, though.

File: jobs/build/ocp4/Jenkinsfile (hunk 1 above)
Comment (jupierce) [reply jobs/build/ocp4/Jenkinsfile]: This is now done in `assertBuildPermitted`

File: jobs/build/ocp4/build.groovy (hunk 2 above)
Comment (jupierce) [reply jobs/build/ocp4/build.groovy]: I put it in buildlib with the same end goal - since it was doozer / build based.

**Comment by sosiouxme** [comment]
thanks!

**Pull request merged**